examples/summarize_scoring_using_bedrock.ipynb

```diff
@@ -0,0 +1,374 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "68397145-e13d-4e6d-bb0a-35944409c333",
+   "metadata": {},
+   "source": [
+    "# Summarization Scoring using AWS Bedrock\n",
+    "\n",
+    "In this example notebook, we will be comparing how different models in AWS Bedrock perform at summarization tasks using **Arthur Bench**. \n",
+    "\n",
+    "The overall summarization comparison is setup as a **Bench TestSuite**, and each model is compared head-to-head against every other model\n",
+    "in a **Bench TestRun**. Bench provides the ability to view these comparisons in the provided User Interface, as well as access statistics\n",
+    "from the Test and TestRuns themselves for further analysis. For example, in the notebook below we provide a means for using the ELO Scoring\n",
+    "Algorithm to determine which model performs best at this summarization task. \n",
+    "                                                                                                                 \n",
+    "The task is to summarize 49 News Articles, and comparison is done using ChatGPT 3.5 Turbo (see summary_quality.py)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c97dda90-22a7-4a8d-8cf5-e2948d35f8af",
+   "metadata": {},
+   "source": [
+    "## 1. Setup AWS Bedrock Client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d375f148-b8f3-4a6d-8a10-f4e96e97ce90",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\"\"\"\n",
+    "Authentication is handled using the AWS_PROFILE environment variable. Check the AWS Boto3 documentation and the provided\n",
+    "utility library for connecting to Bedrock for additional information\n",
+    "\"\"\"\n",
+    "from bedrock_client import client\n",
+    "\n",
+    "bedrock_runtime = client.get_bedrock_client(region=\"us-east-1\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11b3eeae-afa0-474b-974b-bcaa716d83c8",
+   "metadata": {},
+   "source": [
+    "## 2. Load the data and prepare it for inferencing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cfee14bb-017d-4057-a406-1f0cc60e5bee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import csv\n",
+    "\n",
+    "articles = []\n",
+    "\n",
+    "with open(\"data/news_summary/example_summaries.csv\", \"r\") as f:\n",
+    "    dr = csv.DictReader(f)\n",
+    "    for row in dr: \n",
+    "        articles.append(row[\"input_text\"])\n",
+    "\n",
+    "len(articles)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ff9d4a64-1fc0-493c-b7d5-556d4174313e",
+   "metadata": {},
+   "source": [
+    "## 3. Generate inferences (summaries) for the articles"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a6d18def-3fe4-4eba-aacc-0882cf123bd8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
I think this refactored inference code is a bit easier
`def bedrock_invoke(model_id : str, json_request : str)
    modelId = model_id  
    accept = "application/json"
    contentType = "application/json"
    response = bedrock_runtime.invoke_model(
        body=json_request, modelId=modelId, accept=accept, contentType=contentType
    )
    response_body = json.loads(response.get("body").read())
    return response_body

def generate_summary_from_llama(model_id, article): 
    json_request = json.dumps({"prompt": summary_prompt.format(article)})
    response_body = bedrock_invoke(model_id, json_request)
    return response_body.get("generation")

def generate_summary_from_titan(model_id, article):
    json_request = json.dumps({"inputText": summary_prompt.format(article)})
    response_body = bedrock_invoke(model_id, json_request)
    return response_body.get("results")[0].get("outputText")

def generate_summary_from_claude(model_id, article):
    json_request = json.dumps({
        "prompt": f"Human:\n{summary_prompt.format(article)}\nAssistant:\n",
        "max_tokens_to_sample": 4000,  # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-claude.html
    })
    response_body = bedrock_invoke(model_id, json_request)
    return response_body.get("completion")    `

Reply:
i think this refactored inference code is a bit easier

Reply:
Happy to move to a python-first approach (vs. using a notebook) if that's the convention we've been sticking to. 

The things that's missing from the refactor above is defining the models to use, and then generating the unique 2-combinations of models, which is important for the ELO scoring, but based on how you refactored, this would be something that'd be defined by the client of this library. 

Ultimately, the goal with this MR was to publishing something in the repo that highlighted how someone would use Bench to compare models hosted in Bedrock. From that perspective, I think a notebook is an easier interface to get started with (than a python library which has to be invoked/executed each time a change is made). 

For this sort of use-case, would you still advocate for a library? And if so, why?

Reply:
notebook is ok as is